File: server/Models/Project.cs
using System.Collections.Generic;
using my_web_server.DTOs;

namespace my_web_server.Models
{
    public class Project
    {
        public Project() { }

        public Project(ProjectDto projectDto)
        {
            Name = projectDto.Name;
            TechStack = projectDto.TechStack;
            Description = projectDto.Description;
            ShowcaseLink = projectDto.ShowcaseLink;
            RepositoryLink = projectDto.RepositoryLink;
            PreviewLink = projectDto.PreviewLink;
        }

        public int Id { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public string? ShowcaseLink { get; set; }
        public string? RepositoryLink { get; set; }
        public string? PreviewLink { get; set; }
        public List<string>? TechStack { get; set; }
    }
}

File: client/src/sanity/schemaTypes/project.ts
import { defineType } from "sanity";

export default defineType({
    name: "project",
    title: "Project",
    type: "document",
    fields: [
        {
            name: "name",
            title: "Name",
            type: "string",
            validation: (Rule) => Rule.required(),
        },
        {
            name: "slug",
            title: "Slug",
            type: "slug",
            options: { source: "name", maxLength: 96 },
        },
        {
            name: "description",
            title: "Description",
            type: "array",
            of: [{ type: "block" }],
        },
        { name: "repositoryLink", title: "Repository Link", type: "url" },
        { name: "previewLink", title: "Preview Link", type: "url" },
        { name: "showcaseLinkMp4", title: "Showcase Link MP4", type: "url" },
        { name: "showcaseLinkGif", title: "Showcase Link GIF", type: "url" },
        {
            name: "techStack",
            title: "Tech Stack",
            type: "array",
            of: [{ type: "string" }],
        },

        { name: "startDate", title: "Start Date", type: "date" },
        { name: "endDate", title: "End Date", type: "date" },
    ],
});
